pkg/oci/load_balancer_security_lists.go

```diff
@@ -228,6 +233,26 @@ func getNodeIngressRules(securityList *baremetal.SecurityList, lbSubnets []*bare
 			desired.Delete(rule.Source)
 			continue
 		}
+
+		if rule.TCPOptions.DestinationPortRange.Min == port && rule.TCPOptions.DestinationPortRange.Max == port {
```

Comment:
This if statement is redundant isn't it? We're checking that min/max = port above.

Reply:
I guess its not totally redundant as there's the case where only the min (or the max) matches, but the other doesn't, which would pass over the above check, but not the 2nd one. However, thinking about it again, maybe the logic on the above check should be this instead:
```
         if rule.TCPOptions == nil || rule.TCPOptions.SourcePortRange != nil || rule.TCPOptions.DestinationPortRange == nil ||
             rule.TCPOptions.DestinationPortRange.Min != port ||
             rule.TCPOptions.DestinationPortRange.Max != port {
             // this rule doesn't apply to this service so nothing to do but keep it
...
```
This way, we would only take ownership of (and potentially delete) rules where both the min and max match, and if only one matches, we assume the rule isn't ours thus should be left alone. In this case, the 2nd check then does become totally redundant. How does this sound to you? (as to be honest this logic is making my head hurt!)